components/jupyter-web-app/frontend/src/app/resource-form/volume/volume.component.html

```diff
@@ -1,42 +1,49 @@
 <div *ngIf="volume" [formGroup]="volume" class="volume-wrapper">
-  <mat-form-field appearance="outline" id="type">
-    <mat-label>Type</mat-label>
-    <mat-select formControlName="type">
-      <mat-option
+ 
+  <!-- Volume Type Input -->
+    <mat-form-field appearance="outline" id="type">
+      <mat-label>Type</mat-label>
+      <mat-select  (selectionChange)="selectType($event)" formControlName="type">
+        <mat-option *ngFor="let i of types"
         [disabled]="newTypeIsDisabled()"
         [matTooltip]="newTypeTooltip()"
-        value="New"
-        >New</mat-option
-      >
-      <mat-option value="Existing">Existing</mat-option>
-    </mat-select>
-  </mat-form-field>
+        value={{i}}>{{i}}</mat-option>
```

Comment:
```suggestion
        value="{{i}}">{{i}}</mat-option>
```

Reply:
Will test this and commit again - Definitely not an expert on right way to code this.

Reply:
Submitted the commit again after testing.